vochain/indexer/archive.go

```diff
@@ -0,0 +1,138 @@
+package indexer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"go.vocdoni.io/dvote/data"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/types"
+	indexerdb "go.vocdoni.io/dvote/vochain/indexer/db"
+	"go.vocdoni.io/dvote/vochain/indexer/indexertypes"
+	"go.vocdoni.io/dvote/vochain/results"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+const (
+	marxArchiveFileSize     = 1024 * 100 // 100KB
+	timeoutArchiveRetrieval = 60 * time.Second
+	archiveFetchInterval    = 20 * time.Minute
+	archiveFileNameSize     = types.ProcessIDsize * 2 // 64 hex chars
+)
+
+// ArchiveProcess is the struct used to store the process data in the archive.
+type ArchiveProcess struct {
+	ChainID     string                `json:"chainId,omitempty"`
+	ProcessInfo *indexertypes.Process `json:"process"`
+	Results     *results.Results      `json:"results"`
+	StartDate   *time.Time            `json:"startDate,omitempty"`
+	EndDate     *time.Time            `json:"endDate,omitempty"`
+}
+
+// ImportArchive imports an archive list of processes into the indexer database.
+func (idx *Indexer) ImportArchive(archive []*ArchiveProcess) error {
+	tx, err := idx.readWriteDB.Begin()
+	if err != nil {
+		panic(err) // shouldn't happen, use an error return if it ever does
+	}
```

Comment:
you're missing a `defer tx.Discard()` here, otherwise if this func fails you leak the transaction without committing

Reply:
Do you mean tx.Rollback()

Reply:
yes sorry